dotnet/src/dotnetcore/Providers/Messaging/GXAzureQueue/AzureQueue.cs

```diff
@@ -0,0 +1,413 @@
+		{
+			_queueName = queueName;
+			_connectionString = connectionString;
+		}
+
+		//public AzureQueue(Uri uri, TokenCredential tokenCredential)
+		//{
+		//_queueClient = new QueueClient(uri, tokenCredential);
+		//}
+
+		public bool GetMessageFromException(Exception ex, SdtMessages_Message msg)
+		{
+			try
+			{
+				Azure.RequestFailedException az_ex = (Azure.RequestFailedException)ex;
+				msg.gxTpr_Id = az_ex.ErrorCode;
+				msg.gxTpr_Description = az_ex.Message; 
+				return true;
+			}
+			catch (Exception)
+			{
+				return false;
+			}
+		}
+
+		/// <summary>
+		/// Get the approximate number of messages in the queue
+		/// </summary>
+		/// <param name="success"></param>
+		/// <returns></returns>
+		public int GetQueueLength(out bool success)
+		{
+			int cachedMessagesCount = 0;
+			success = false;
+			if (_queueClient is QueueClient && _queueClient.Exists())
+			{
+				QueueProperties properties = _queueClient.GetProperties();
+
+				// Retrieve the cached approximate message count.
+				cachedMessagesCount = properties.ApproximateMessagesCount;
+				success = true;
+			}
+			return cachedMessagesCount;
+		}
+
+		/// <summary>
+		/// Deletes all messages from a queue.
+		/// </summary>
+		public void Clear(out bool success)
+		{
+			success = false;
+			if (_queueClient is QueueClient && _queueClient.Exists())
+			{
+				Azure.Response result = _queueClient.ClearMessages();
+				success = !result.IsError;
+			}
+		}
+
+		/// <summary>
+		/// Permanently removes the first message dequeued.
+		/// </summary>
+
+		public MessageQueueResult DeleteMessage(out bool success)
+		{
+			success=false;
+			MessageQueueResult messageQueueResult = new MessageQueueResult();
+			if (_queueClient is QueueClient && _queueClient.Exists())
+			{
+				Azure.Response<QueueMessage> receivedMessage = _queueClient.ReceiveMessage();
+
+				if ((receivedMessage != null) && (!receivedMessage.GetRawResponse().IsError) && (receivedMessage.Value != null))
+				{
+					Azure.Response deleteResult = _queueClient.DeleteMessage(receivedMessage.Value.MessageId, receivedMessage.Value.PopReceipt);
+
+					success = !deleteResult.IsError;
+					if (success)
+					{
+						return (AzQueueMessageToMessageQueueResult(receivedMessage.Value, MessageQueueResultStatus.Deleted));
+					}
+				}
+			}
+			return messageQueueResult;
+		}
+
+		/// <summary>
+		/// Deletes permanently the messages given on the list.
+		/// </summary>
+		
+		public IList<MessageQueueResult> DeleteMessages(List<string> messageHandleId, MessageQueueOptions messageQueueOptions,out bool success)
+		{
+			success = false;
+			IList<MessageQueueResult> messageQueueResults = new List<MessageQueueResult>();
+			if (_queueClient is QueueClient && _queueClient.Exists())
+			{
+				QueueProperties properties = _queueClient.GetProperties();
+
+				QueueMessage[] receivedMessages = _queueClient.ReceiveMessages(messageQueueOptions.MaxNumberOfMessages);
```

Comment:
Does  not exists a direct way to delete message just with their messageId? Seems unreliable to need to consume all the queue in order to be able to delete. The only case is see for this is FIFO Queues.

Reply:
No, I need the PopReceipt of the message.